Add inline example of PyTorch pyfunc usage (#7121)
mlflow/mlflow: docs/source/models.rst

    In case of multi gpu training, ensure to save the model only with global rank 0 gpu. This avoids
    logging multiple copies of the same model.

> PyTorch pyfunc usage
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
PyTorch pyfunc usage
~~~~~~~~~~~~~~~~~~~~
```

For a minimal PyTorch model, an example configuration for the pyfunc predict() method is:

.. code-block:: py
    
    import numpy as np
    import mlflow
    import torch
    from torch import nn

>     class Net(nn.Module):
>         def __init__(self):
>             super().__init__()
>             self.l1 = nn.Linear(6, 1)
> 
>         def forward(self, x):
>             return self.l1(x)
> 
> 
>     net = Net()

Reviewer: ```suggestion
    net = nn.Linear(6, 1)
```

Does this work?

    loss_function = nn.L1Loss()
    optimizer = torch.optim.Adam(net.parameters(), lr=1e-4)

    X = torch.randn(6)
    y = torch.randn(1)

    epochs = 5
    for epoch in range(epochs):
        optimizer.zero_grad()
        outputs = net(X)

        loss = loss_function(outputs, y)
        loss.backward()

        optimizer.step()

>     with mlflow.start_run() as run:
>         mlflow.pytorch.save_model(net, "net")
> 
>     pytorch_pyfunc = mlflow.pyfunc.load_model(model_uri="net")
> 

Reviewer: ```suggestion
    with mlflow.start_run() as run:
        model_info = mlflow.pytorch.log_model(net, "model")

    pytorch_pyfunc = mlflow.pyfunc.load_model(model_uri=model_info.model_uri)

```

>     predictions = pytorch_pyfunc.predict(np.array([3.4, -1.2, 4.8, 7.1, 9.4, -6.9], np.float32))

Reviewer: ```suggestion
    predictions = pytorch_pyfunc.predict(np.array([3.4, -1.2, 4.8, 7.1, 9.4, -6.9], np.float32))
    print(predictions)
```
